experimental/parser/legalize_type.go

```diff
@@ -16,11 +16,118 @@ package parser
 
 import (
 	"github.com/bufbuild/protocompile/experimental/ast"
+	"github.com/bufbuild/protocompile/experimental/ast/predeclared"
 	"github.com/bufbuild/protocompile/experimental/internal/taxa"
+	"github.com/bufbuild/protocompile/experimental/report"
+	"github.com/bufbuild/protocompile/internal/ext/iterx"
 )
 
 func legalizeMethodParams(p *parser, list ast.TypeList, what taxa.Noun) {
+	if list.Len() != 1 {
+		p.Errorf("expected exactly one type in %s, got %d", what, list.Len()).Apply(
+			report.Snippet(list),
+		)
+		return
+	}
+
+	ty := list.At(0)
+	switch ty.Kind() {
+	case ast.TypeKindPath:
+		// Allow all path types. We don't know if this is a message or not yet;
+		// figuring that out is a problem for IR construction.
```

Comment:
We should not allow a path that has slashes or internal paths (i.e. extension names).